core/src/main/scala/chisel3/internal/MonoConnect.scala

```diff
@@ -372,7 +372,7 @@ private[chisel3] object MonoConnect {
   }
 
   // This function (finally) issues the connection operation
-  private def issueConnect(sink: Element, source: Element)(implicit sourceInfo: SourceInfo): Unit = {
+  private def issueConnect(sink: BaseType, source: BaseType)(implicit sourceInfo: SourceInfo): Unit = {
```

Comment:
Shouldn't this throw an error at this point if it's not actually a Data?

Reply:
That's a fair point, I can add a runtime error if the type's aren't Element. I have a patch staged after this which will handle Property as well as Element, but to be defensive, I think it would make sense to always have an error in the base case rather than assuming issuing a connect statement is correct.

Reply:
I've added a check that both source and sink are Element types in the most recent push. With the current public APIs it should be impossible for this to fail, so I raised an exception with an "Internal Error!" message. I was trying to write a test for this, but again, I don't think it is possible to exercise this behavior with the current public APIs.

Reply:
I don't really see what value this is adding. I get that you want to get some reuse here for Properties but considering this function is only 4 lines long, is there really much to reuse here vs. just writing a specific function for Properties?

Reply:
Oh I see, this is called by `elemConnect`, this is still weird though and I'm not sure how I feel about it.

Reply:
Yeah, I guess the alternative is to factor out all the checks in elemConnect to a helper that can be used for Properties. Then this PR just becomes factoring out that helper. Is that better?